### ISSUE TITLE
Test failure on ghc 9.10

### ISSUE THREAD

**Issue by eldritch-cookie** [Soostone/katip#155]
[katip-testsuite.log](https://github.com/user-attachments/files/17295935/katip-testsuite.log)

**Comment by eldritch-cookie** [comment]
yes i can run the testsuite on 0.8.8.2 it fails again [katip-0.8.8.2-build.log](https://github.com/user-attachments/files/17350049/katip-0.8.8.2-build.log)
the test that failed is  the one on line 163 https://github.com/Soostone/katip/blob/af85a1fdd39b132feb353db12080a8f59476a971/katip/test/Katip/Tests.hs#L163 not a quickcheck test, the error is a call to head so it probably is caused by https://github.com/Soostone/katip/blob/af85a1fdd39b132feb353db12080a8f59476a971/katip/test/Katip/Tests.hs#L168 but i have no idea why loggedItems is an empty list.

**Comment by 9999years** [comment]
Comes from here:

https://github.com/Soostone/katip/blob/af85a1fdd39b132feb353db12080a8f59476a971/katip/src/Katip/Core.hs#L1155

**Comment by MichaelXavier** [comment]
@9999years Do you understand the issue well enough to write up a fix? I'd be more than happy to merge and release it but I don't have much time to chase this down.

**Comment by 9999years** [comment]
@MichaelXavier I think factoring the tests out into a separate package whose name doesn't start with `katip-` would work.

**Comment by juhp** [comment]
Also failed for Stackage Nightly

**Comment by maksar** [comment]
No longer valid since https://github.com/Soostone/katip/pull/159/files
